packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * @flow
+ */
+
+import {
+  type EventSystemFlags,
+  IS_PASSIVE,
+  PASSIVE_NOT_SUPPORTED,
+} from 'events/EventSystemFlags';
+import type {AnyNativeEvent} from 'events/PluginModuleType';
+import {EventComponent} from 'shared/ReactWorkTags';
+import type {ReactEventResponder} from 'shared/ReactTypes';
+import invariant from 'shared/invariant';
+import warning from 'shared/warning';
+import type {DOMTopLevelEventType} from 'events/TopLevelEventTypes';
+import accumulateInto from 'events/accumulateInto';
+import SyntheticEvent from 'events/SyntheticEvent';
+import {runEventsInBatch} from 'events/ReactGenericBatching';
+import {interactiveUpdates} from 'events/ReactGenericBatching';
+import type {Fiber} from 'react-reconciler/src/ReactFiber';
+
+import {getClosestInstanceFromNode} from '../client/ReactDOMComponentTree';
+
+// We track the active component fibers so we can traverse through
+// the fiber tree and find the relative current fibers. We need to
+// do this because an update might have switched an event component
+// fiber to its alternate fiber.
+export const currentEventComponentFibers: Set<Fiber> = new Set();
+
+// Event responders provide us an array of target event types.
+// To ensure we fire the right responders for given events, we check
+// if the incoming event type is actually relevant for an event
+// responder. Instead of doing an O(n) lookup on the event responder
+// target event types array each time, we instead create a Set for
+// faster O(1) lookups.
+export const eventResponderValidEventTypes: Map<
+  ReactEventResponder,
+  Set<DOMTopLevelEventType>,
+> = new Map();
+
+type EventListener = (event: SyntheticEvent) => void;
+
+// TODO add context methods for dispatching events
+function DOMEventResponderContext(
+  topLevelType: DOMTopLevelEventType,
+  nativeEvent: AnyNativeEvent,
+  nativeEventTarget: EventTarget,
+  eventSystemFlags: EventSystemFlags,
+) {
+  this.event = nativeEvent;
+  this.eventType = topLevelType;
+  this.eventTarget = nativeEventTarget;
+  this._flags = eventSystemFlags;
+  this._fiber = null;
+  this._responder = null;
+  this._discreteEvents = [];
+  this._nonDiscreteEvents = [];
+}
+
+DOMEventResponderContext.prototype.isPassive = function(): boolean {
```

Comment:
`context` is passed to the event responder module so that the user can call `isPassive` etc without having to know about the React internals. I thought about exposing these functions via a public API on React, but that felt like over-kill, especially as the responder modules are external to React.

Reply:
Cool. I don't fully see how everything will fit together. We can come back to this later once more stuff is in place.